project_task_subtask/models/project_task_subtask.py

```diff
@@ -0,0 +1,69 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, fields, api
+from openerp.tools import html_escape as escape
+
+
+class ProjectTaskSubtask(models.Model):
+    _name = "project.task.subtask"
+    state = fields.Selection([('done', 'Done'),
+                              ('todo', 'Todo'),
+                              ('cancelled', 'Cancelled')],
+                             'Status', required=True, copy=False, default='todo')
+    name = fields.Char(required=True, string="Description")
+    reviewer_id = fields.Many2one('res.users', 'Reviewer', select=True, required=True)
+    project_id = fields.Many2one("project.project", related='task_id.project_id', store=True)
+    user_id = fields.Many2one('res.users', 'Assigned to', select=True, required=True)
+    task_id = fields.Many2one('project.task', 'Task', ondelete='cascade', required=True, select="1")
+
+    @api.multi
+    def write(self, vals):
+        result = super(ProjectTaskSubtask, self).write(vals)
+        for r in self:
+            if vals.get('state'):
+                r.task_id.send_subtask_email(r.name, r.state)
+        return result
+
+
+class Task(models.Model):
+    _inherit = "project.task"
+    subtask_ids = fields.One2many('project.task.subtask', 'task_id', 'Subtask')
+    kanban_subtasks = fields.Text(compute='_compute_kanban_subtasks')
+
+    @api.multi
+    def _compute_kanban_subtasks(self):
+        for record in self:
+            result_string = '<ul>'
+            for subtask in record.subtask_ids:
+                if subtask.state == 'todo' and record.env.user == subtask.user_id:
+                    tmp_string1 = 'From {0}: {1}'.format(subtask.reviewer_id.name, subtask.name)
+                    result_string += '<li>{}</li>'.format(escape(tmp_string1))
+            for subtask in record.subtask_ids:
+                if subtask.state == 'todo' and record.env.user == subtask.reviewer_id:
+                    tmp_string2 = 'To {0}: {1}'.format(subtask.user_id.name, subtask.name)
+                    result_string += '<li>{}</li>'.format(escape(tmp_string2))
+            record.kanban_subtasks = result_string + '</ul>'
+
+    @api.multi
+    def send_subtask_email(self, subtask_name, subtask_state):
+        for r in self:
+            template = self.env.ref('project_task_subtask.email_template_subtask_changed')
+            email_ctx = {
+                'default_model': 'project.task',
+                'default_res_id': r.id,
+                'default_use_template': bool(template),
+                'default_template_id': template.id,
+                'subtask_name': subtask_name,
+                'subtask_state': subtask_state,
```

Comment:
It is a technical name for now. Fix it to use one from Selection value, e.g. "Todo" instead of "todo". 

You probably need to move Selection list to a global var

```
SUBTASK_STATES = [('done', 'Done'),
                              ('todo', 'Todo'),
                              ('cancelled', 'Cancelled')]
                              

+    state = fields.Selection(SUBTASK_STATES,
 +                             'Status', required=True, copy=False, default='todo')

+                'subtask_state': SUBTASK_STATES[subtask_state],

```